plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java

```diff
@@ -107,10 +107,12 @@ public Object doCall(TaskExecutionGraph graph) {
 					.all(task -> task.dependsOn(rootCheckTask));
 		}
 
-		// clear spotless' cache when the user does a clean
+		// clear spotless' cache when the user does a clean, but only after any spotless tasks
+		Task clean = project.getTasks().getByName(BasePlugin.CLEAN_TASK_NAME);
+		clean.doLast(unused -> SpotlessCache.clear());
 		project.getTasks()
-				.matching(task -> task.getName().equals("clean"))
-				.all(task -> task.doLast(unused -> SpotlessCache.clear()));
+				.matching(task -> task instanceof SpotlessTask)
+				.all(task -> task.mustRunAfter(clean));
```

Comment:
Gradle has a method (at least in the newer versions) that's something like:
```java
project.getTasks().withType(SpotlessTask.class).all(task -> task.mustRunAfter(clean));
```

Reply:
Awesome, thanks!